src/ServiceControl/Recoverability/Retrying/Infrastructure/ReturnToSenderDequeuer.cs

```diff
@@ -318,24 +318,26 @@ public void Init(Address address)
 
         internal class DequeueMessagesWrapper : IDequeueMessages
         {
-            private readonly IDequeueMessages _realDequeuer;
-            private int maximumConcurrencyLevel;
-            private int disposeSignaled;
+            IDequeueMessages realDequeuer;
+            int maximumConcurrencyLevel;
+            object startStopLock = new object();
 
             public DequeueMessagesWrapper(IDequeueMessages realDequeuer)
             {
-                _realDequeuer = realDequeuer;
+                this.realDequeuer = realDequeuer;
             }
 
             public void Init(Address address, TransactionSettings transactionSettings, Func<TransportMessage, bool> tryProcessMessage, Action<TransportMessage, Exception> endProcessMessage)
             {
-                _realDequeuer.Init(address, transactionSettings, tryProcessMessage, endProcessMessage);
+                realDequeuer.Init(address, transactionSettings, tryProcessMessage, endProcessMessage);
             }
 
             public void StartInternal()
             {
-                Interlocked.Exchange(ref disposeSignaled, 0);
```

Comment:
@SzymonPobiega Do you know what this was even used for?

Reply:
It was used to make sure that `Stop` is not called concurrently. So in the old code `Start` and `Stop` could overlap but two `Stops` could not as one of the `Stops` would detect the flag signalled.